source/dubregistry/dbcontroller.d

```diff
@@ -324,18 +341,10 @@ class DbController {
 		import std.math : round;
 
 		if (!query.strip.length) {
-			return m_packages.find()
-				.sort(["stats.score": 1])
-				.map!(deserializeBson!DbPackage)
-				.array;
+			return pkgCache;
 		}
 
-		auto pkgs = m_packages
-			.find(["$text": ["$search": query]], ["textScore": bson(["$meta": "textScore"])])
-			.sort(["textScore": bson(["$meta": "textScore"])]) // sort to only keep most relevant results
-			.limit(50) // limit irrelevant sort results (fixes #341)
-			.map!(deserializeBson!DbPackage)
-			.array;
+		DbPackage[] pkgs = pkgCache.filter!(p => p.name.canFind(query)).array;
```

Comment:
instead of doing the filtering here I think it would be more efficient doing it on the MongoDB side using `$regex`

However building a better text index would be the much better solution.

Reply:
Feel free to implement it